src/main/java/com/sap/sgs/phosphor/fosstars/model/score/oss/ProjectActivityScore.java

```diff
@@ -26,45 +29,76 @@
+   * where m is a number of contributors in the last three months,
+   * and A = MAX_CONTRIBUTORS_SUB_SCORE - MAX_CONTRIBUTORS_SUB_SCORE_SHIFT.
+   * The function belongs to the interval [0.0, MAX_CONTRIBUTORS_SUB_SCORE) when m >= 0.
+   */
+  private static final Logistic LOGISTIC_FOR_NUMBER_OF_CONTRIBUTORS
+      = logistic(MAX_CONTRIBUTORS_SUB_SCORE - MAX_CONTRIBUTORS_SUB_SCORE_SHIFT, 0.4, 4);
+
+  /**
+   * A final function which is used to convert a number of contributors to a sub-score:
+   * <pre>y(m) = LOGISTIC_FOR_NUMBER_OF_CONTRIBUTORS(m) - MAX_CONTRIBUTORS_SUB_SCORE_SHIFT</pre>.
+   */
+  private static final UnivariateFunction NUMBER_OF_CONTRIBUTORS_SUB_SCORE =
+      m -> LOGISTIC_FOR_NUMBER_OF_CONTRIBUTORS.value(m) + MAX_CONTRIBUTORS_SUB_SCORE_SHIFT;
+
+  /**
+   * Returns a logistic function y(x) = a / (1 + e ^ k(b - x)).
+   */
+  private static Logistic logistic(double a, double k, double b) {
+    return new Logistic(a, b, k, 1, 0, 1);
+  }
+
+  /**
+   * A description of the score.
+   */
   private static final String DESCRIPTION;
 
   static {
     StringBuilder sb = new StringBuilder();
 
     sb.append("The score is based on number of commits and contributors.\n");
-    sb.append("Here is how a number of commits contributes to the score:\n");
-    sb.append(String.format("%d -> %2.2f (min), ", 0, LOGISTIC_FOR_NUMBER_OF_COMMITS.value(0)));
-    sb.append(String.format("%d -> %2.2f, %d -> %2.2f, ",
-        MathHelper.invert(LOGISTIC_FOR_NUMBER_OF_COMMITS, 0, 1000, 2.5, 0.01), 2.5,
-        MathHelper.invert(LOGISTIC_FOR_NUMBER_OF_COMMITS, 0, 1000, 5.0, 0.01), 5.0));
-    sb.append(String.format("%d -> %2.2f (max)", 500, LOGISTIC_FOR_NUMBER_OF_COMMITS.value(500)));
+    sb.append(String.format(
+        "Here is how a number of commits contributes to the score (up to %2.2f):\n",
+        MAX_COMMITS_SUB_SCORE));
+    sb.append(printInvertedValues(
+        LOGISTIC_FOR_NUMBER_OF_COMMITS::value, 0, 1000,
+        0.1, MAX_COMMITS_SUB_SCORE * 0.5, MAX_COMMITS_SUB_SCORE * 0.9));
 
     sb.append("\n");
-    sb.append("Here is how a number of contributors contributes to the score:\n");
-    sb.append(String.format("%d -> %2.2f (min), ",
-        0, LOGISTIC_FOR_NUMBER_OF_CONTRIBUTORS.value(0)));
-    sb.append(String.format("%d -> %2.2f, %d -> %2.2f, ",
-        MathHelper.invert(LOGISTIC_FOR_NUMBER_OF_CONTRIBUTORS, 0, 50, 1.5, 0.01), 1.5,
-        MathHelper.invert(LOGISTIC_FOR_NUMBER_OF_CONTRIBUTORS, 0, 100, 2.5, 0.01), 2.5));
-    sb.append(String.format("%d -> %2.2f (max)",
-        500, LOGISTIC_FOR_NUMBER_OF_CONTRIBUTORS.value(500)));
+    sb.append(String.format(
+        "Here is how a number of contributors contributes to the score (up to %2.2f):\n",
+        MAX_CONTRIBUTORS_SUB_SCORE));
+    sb.append(printInvertedValues(
+        NUMBER_OF_CONTRIBUTORS_SUB_SCORE::value, 0, 200,
+        0.1, MAX_CONTRIBUTORS_SUB_SCORE * 0.5, MAX_CONTRIBUTORS_SUB_SCORE * 0.9));
 
     DESCRIPTION = sb.toString();
   }
```

Comment:
There is a line which we missed I guess:
```
int n = numberOfCommits.get();
if (numberOfCommits.get() == 0) {
      return 0;	   
}
```
to 

```
int n = numberOfCommits.get();
if (n == 0) {
      return 0;	   
}
```

Reply:
Similarly 

```
int n = numberOfContributors.get();
if (numberOfContributors.get() == 0) {
      return 0;	   
}
```
to 
```
int n = numberOfContributors.get();
if (n == 0) {
      return 0;	   
}
```

Reply:
Okay.